### PR TITLE
Feature/dummy data

### PR DESCRIPTION
**Pull request by ali3654**
Co-authored-by: Thabb <49516328+Thabb@users.noreply.github.com>
Co-authored-by: kosch104 <konsicrafter@web.de>
Co-authored-by: Segelzwerg <marcel.haas@hhu.de>

### CODE REVIEW

File: src/main/java/mops/DummyDataApplication.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import mops.persistence.permission.DirectoryPermissions;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+@Configuration
+public class DummyDataApplication {
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]: this class should be called DummyDataSeeding

File: src/main/java/mops/DummyDataApplication.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import mops.persistence.file.FileInfo;
+import mops.persistence.permission.DirectoryPermissions;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+@Configuration
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]: needs `@Profile("dev")`

File: src/main/java/mops/DummyDataApplication.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+public class DummyDataApplication {
+
+
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
```

File: src/main/java/mops/DummyDataApplication.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    @Bean
+    ApplicationRunner init(DirectoryRepository directoryRepository, FileInfoRepository fileInfoRepository, DirectoryPermissionsRepository directoryPermissionsRepository) {
+        return args -> {
+            DirectoryPermissions directoryPermissions = DirectoryPermissions.builder()
+                    .entry("Admin", true, true, true)
+                    .build();
+
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
```

File: src/main/java/mops/DummyDataApplication.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+@Configuration
+public class DummyDataApplication {
+
+
+    @Bean
+    ApplicationRunner init(DirectoryRepository directoryRepository, FileInfoRepository fileInfoRepository, DirectoryPermissionsRepository directoryPermissionsRepository) {
+        return args -> {
+            DirectoryPermissions directoryPermissions = DirectoryPermissions.builder()
+                    .entry("Admin", true, true, true)
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
                    .entry("admin", true, true, true)
```

File: src/main/java/mops/DummyDataApplication.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                    .build();
+
+
+            directoryPermissions = directoryPermissionsRepository.save(directoryPermissions);
+
+            Directory directoryParent = Directory.builder()
+                    .name("Root")
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
                    .name("")
```

File: src/main/java/mops/DummyDataApplication.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                    .groupOwner(100L)
+                    .permissions(directoryPermissions.getId())
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
                    .permissions(directoryPermissions)
```

File: src/main/java/mops/DummyDataApplication.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+                    .build();
+
+            directoryParent = directoryRepository.save(directoryParent);
+
+            Directory directoryChild = Directory.builder()
+                    .name("Child")
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]: there is `fromParent` method

File: src/main/java/mops/DummyDataApplication.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                    .parent(directoryParent.getId())
+                    .groupOwner(100L)
+                    .permissions(directoryPermissions.getId())
+                    .build();
+
+            directoryChild = directoryRepository.save(directoryChild);
+
+            FileInfo fileInfoParent = FileInfo.builder()
+                    .name("Test1")
+                    .directory(directoryParent.getId())
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
                    .directory(directoryParent)
```

File: src/main/java/mops/DummyDataApplication.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                    .type("PDF")
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
                    .type("application/pdf")
```

File: src/main/java/mops/DummyDataApplication.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+                    .size(2000L)
+                    .owner("Hebert")
+                    .tag("Test")
+                    .build();
+
+            FileInfo fileInfoChild = FileInfo.builder()
+                    .name("Test2")
+                    .directory(directoryChild.getId())
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
                    .directory(directoryChild)
```

File: src/main/java/mops/DummyDataApplication.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+                    .type("PNG")
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
                    .type("image/png")
```

File: src/main/java/mops/DummyDataApplication.java [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+            Optional<DirectoryPermissions> directoryPermissionsFind = directoryPermissionsRepository.findById(directoryPermissions.getId());
+            Optional<Directory> directoryFind = directoryRepository.findById(directoryParent.getId());
+            Optional<FileInfo> fileInfoFind = fileInfoRepository.findById(fileInfoParent.getId());
+            System.out.println(directoryPermissionsFind);
+            System.out.println(directoryFind);
+            System.out.println(fileInfoFind);
+
+
+            ;
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
```

File: src/main/java/mops/DummyDataApplication.java [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+        };
+    }
+
+
```
Comment (iTitus) [inline src/main/java/mops/DummyDataApplication.java]:
```suggestion
```

File: src/main/java/mops/DummyDataApplication.java (hunk 6 above)
Comment (Thabb) [reply src/main/java/mops/DummyDataApplication.java]: Warum?

Comment (iTitus) [reply src/main/java/mops/DummyDataApplication.java]: Weil das Root-Dir als Namen `""` hat.

File: src/main/java/mops/DummyDataSeeding.java [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    @Bean
+    @Profile("dev")
+    public ApplicationRunner init(DirectoryRepository directoryRepo, FileInfoRepository fileInfoRepo,
+                                  DirectoryPermissionsRepository directoryPermissionsRepo) {
+        return args -> {
+            final long onehundred = 100L;
+            final long twothousand = 2000L;
+            final long threethousand = 3000L;
+
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
            final long groupId = 100L;
            final long fileSize1 = 2_000L;
            final long fileSize2 = 3_000L;
            final String owner1 = "studentin";
            final String owner2 = "studentin1";
```

File: src/main/java/mops/DummyDataSeeding.java [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+
+            DirectoryPermissions directoryPermissions = DirectoryPermissions.builder()
+                    .entry("admin", true, true, true)
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
                    .entry("admin", true, true, true)
                    .entry("editor", true, true, false)
                    .entry("viewer", true, false, false)
```

File: src/main/java/mops/DummyDataSeeding.java [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+                    .build();
+
+            directoryPermissions = directoryPermissionsRepo.save(directoryPermissions);
+
+            Directory directoryParent = Directory.builder()
+                    .name("")
+                    .groupOwner(onehundred)
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
                    .groupOwner(groupId)
```

File: src/main/java/mops/DummyDataSeeding.java [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+                    .name("Child")
+                    .build();
+
+            directoryChild = directoryRepo.save(directoryChild);
+
+            FileInfo fileInfoParent = FileInfo.builder()
+                    .name("Test1")
+                    .directory(directoryParent)
+                    .type("application/pdf")
+                    .size(twothousand)
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
                    .size(fileSize1)
```

File: src/main/java/mops/DummyDataSeeding.java [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+                    .owner("Hebert")
+                    .tag("Test")
+                    .build();
+
+            FileInfo fileInfoChild = FileInfo.builder()
+                    .name("Test2")
+                    .directory(directoryChild)
+                    .type("image/png")
+                    .size(threethousand)
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
                    .size(fileSize2)
```

File: src/main/java/mops/DummyDataSeeding.java [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+                    .build();
+
+            directoryChild = directoryRepo.save(directoryChild);
+
+            FileInfo fileInfoParent = FileInfo.builder()
+                    .name("Test1")
+                    .directory(directoryParent)
+                    .type("application/pdf")
+                    .size(twothousand)
+                    .owner("Hebert")
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
                    .owner(owner1)
```

File: src/main/java/mops/DummyDataSeeding.java [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+                    .tag("Test")
+                    .build();
+
+            FileInfo fileInfoChild = FileInfo.builder()
+                    .name("Test2")
+                    .directory(directoryChild)
+                    .type("image/png")
+                    .size(threethousand)
+                    .owner("MAX")
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
                    .owner(owner2)
```

File: src/main/java/mops/DummyDataSeeding.java [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+            final String owner1 = "studentin";
+            final String owner2 = "studentin1";
+
+            DirectoryPermissions directoryPermissions = DirectoryPermissions.builder()
+                    .entry("admin", true, true, true)
+                    .entry("editor", true, true, false)
+                    .entry("viewer", true, false, false)
+                    .entry("editor", true, true, true)
+                    .entry("korrektor", true, true, true)
+                    .entry("viewer", true, true, true)
```
Comment (iTitus) [inline src/main/java/mops/DummyDataSeeding.java]:
```suggestion
```
duplicate
the builder should technically crash here

**Pull request merged**